skfp/distances/simpson.py

```diff
@@ -0,0 +1,193 @@
+    Passing all-zero vectors to this function results in a similarity of 0.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    -------
+    similarity : float
+        simpson similarity between vec_a and vec_b.
+
+    References
+    ----------
+    ----------
+    .. [1] `Simpson, E.H.
+       "Measurement of Diversity."
+       Nature 163, 688 (1949).
+       <https://doi.org/10.1038/163688a0>`_
+
+    .. [2] `Deza M.M., Deza E.
+       "Encyclopedia of Distances."
+       Springer, Berlin, Heidelberg, 2009.
+       <https://doi.org/10.1007/978-3-642-00234-2_1>`_
+
+    .. [3] `RDKit documentation
+       <https://www.rdkit.org/docs/source/rdkit.DataStructs.cDataStructs.html>`_
+
+    Examples
+    --------
+    >>> from skfp.distances import simpson_binary_similarity
+    >>> import numpy as np
+    >>> vec_a = np.array([1, 0, 1])
+    >>> vec_b = np.array([1, 0, 1])
+    >>> sim = simpson_binary_similarity(vec_a, vec_b)
+    >>> sim  # doctest: +SKIP
+    1.0
+
+    >>> from skfp.distances import simpson_binary_similarity
+    >>> from scipy.sparse import csr_array
+    >>> vec_a = csr_array([[1, 0, 1]])
+    >>> vec_b = csr_array([[1, 0, 1]])
+    >>> sim = simpson_binary_similarity(vec_a, vec_b)
+    >>> sim  # doctest: +SKIP
+    1.0
+    """
+    _check_nan(vec_a)
+    _check_nan(vec_b)
+
+    if np.sum(vec_a) == 0 == np.sum(vec_b):
+        return 0.0
+
+    if isinstance(vec_a, csr_array) and isinstance(vec_b, csr_array):
+        vec_a_bool = vec_a.astype(bool)
+        vec_b_bool = vec_b.astype(bool)
+        return _simpson_binary_scipy(vec_a_bool, vec_b_bool)
+
+    elif isinstance(vec_a, np.ndarray) and isinstance(vec_b, np.ndarray):
+        vec_a_bool = vec_a.astype(bool)
+        vec_b_bool = vec_b.astype(bool)
+        return _simpson_binary_numpy(vec_a_bool, vec_b_bool)
+
+    else:
+        raise TypeError(
+            f"Both vec_a and vec_b must be of the same type, either numpy.ndarray "
+            f"or scipy.sparse.csr_array, got {type(vec_a)} and {type(vec_b)}"
+        )
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def simpson_binary_distance(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    """
+    Simpson distance for vectors of binary values.
+
+    Computes the Simpson distance for binary data between two input arrays
+    or sparse matrices by subtracting the similarity from 1, using to
+    the formula:
+
+    .. math::
+        dist(vec_a, vec_b) = 1 - sim(vec_a, vec_b)
+
+    The calculated distance falls within the range ``[0, 1]``.
+    Passing all-zero vectors to this function results in a distance of 0.
+
```

Comment:
We also include literature references in distance functions. Please add them here, and also for Rand similarity. Similarity and distance functions generally should have the same literature references.